autonat.go

```diff
@@ -159,17 +157,23 @@ func (as *AmbientAutoNAT) background() {
 	delay := as.config.bootDelay
 
 	var lastAddrUpdated time.Time
-	addrUpdatedChan := as.subAddrUpdated.Out()
-	defer as.subAddrUpdated.Close()
+	subAddrUpdated, _ := as.host.EventBus().Subscribe(new(event.EvtLocalAddressesUpdated))
+	addrUpdatedChan := subAddrUpdated.Out()
+	defer subAddrUpdated.Close()
+
+	subIDOccured, _ := as.host.EventBus().Subscribe(new(event.EvtPeerIdentificationCompleted))
+	IDChan := subIDOccured.Out()
+	defer subIDOccured.Close()
 	defer as.emitReachabilityChanged.Close()
 
 	timer := time.NewTimer(delay)
 	defer timer.Stop()
 	timerRunning := true
+	var peer peer.ID
 
 	for {
 		select {
-		// new connection occured.
+		// new inbound connection.
 		case conn := <-as.inboundConn:
```

Comment:
We should also check "is the remote addr public". We can do this in a followup PR.